libraries/EEPROM/examples/eeprom_crc/eeprom_crc.ino

```diff
@@ -34,16 +34,22 @@ void loop() {
 
 unsigned long eeprom_crc(void) {
 
-  const unsigned long crc_table[17] = {
+  const unsigned long crc_table[16] = {
     0x00000000, 0x1db71064, 0x3b6e20c8, 0x26d930ac,
     0x76dc4190, 0x6b6b51f4, 0x4db26158, 0x5005713c,
     0xedb88320, 0xf00f9344, 0xd6d6a3e8, 0xcb61b38c,
     0x9b64c2b0, 0x86d3d2d4, 0xa00ae278, 0xbdbdf21c
   };
-
+  
   unsigned long crc = ~0L;
 
   for (int index = 0 ; index < EEPROM.length()  ; ++index) {
+    if(index > 16)
+    {
+     Serial.print("\nOut of Bound!");
+     break;
+    }
+     
     crc = crc_table[(crc ^ EEPROM[index]) & 0x0f] ^ (crc >> 4);
     crc = crc_table[(crc ^ (EEPROM[index] >> 4)) & 0x0f] ^ (crc >> 4);
```

Comment:
@thekunalsaini, if you look closely at the code, you see that `index` is used only to index the `EEPROM` itself. The result of that is split into two 4-bit parts (guaranteed using `& 0xf`), which are then used to index `crc_table`. Since only 4 bits are used, the resulting index can never be more than 15, so there is no out of bounds access.

Do you understand what I mean? Or am I missing something here?